calico_containers/pycalico/netns.py

```diff
@@ -180,6 +179,32 @@ def remove_ip_from_ns_veth(container_pid, ip, veth_name_ns):
                        "dev", "%(device)s" % veth_name_ns])
 
 
+def provision_veth(self, endpoint, ns_pid, veth_name_ns):
+    """
+    Create the veth, move into the container namespace, add the IP and
+    set up the default routes.
+
+    Note, the endpoint will not be updated in ETCD. If desired, the user
+    should update the endpoint mac with the mac address provided
+    by the function and then call update_endpoint
+
+    :param endpoint: The endpoint object to provision the veth on
+    :param ns_pid: The PID of the namespace to operate in
+    :param veth_name_ns: The name of the interface in the namespace
+    :return The mac address of the veth as a string
+    """
+    create_veth(endpoint.name, endpoint.temp_interface_name)
+    for ip in endpoint.ipv4_nets | endpoint.ipv6_nets:
```

Comment:
Ooh that's a significant design flaw with this whole approach. For mesos, the API call which creates the interface is passed the desired IP, but the API call which moves the interface into the NS does not - so it is "necessary" in that situation to pull the IP from the endpoint, as the function that calls `provision_veth` will not know what IP address to pass it.

@Symmetric Is there a reason that the libnetwork design approach doesn't bind an IP to an endpoint? Can we?